Update ci-vs-cli.md (#1952)
semgrep/semgrep-docs: docs/kb/semgrep-ci/ci-vs-cli.md

## Pro analysis

> When using `semgrep ci` with Semgrep AppSec Platform, the Semgrep scan reflects the Pro Engine options configured in Semgrep AppSec Platform at https://semgrep.dev/orgs/-/settings. If Pro Engine is enabled, and the scanned code includes [supported Pro languages](/docs/supported-languages/#semgrep-code-language-support), then interfile and interprocedural analysis is performed.

Reviewer: hey @tembalii! I don't necessarily have issues with these changes, but could you please provide context for why this should be deleted? Thanks!

Reviewer 2: Concur, I don't see any reason to remove this.

Author: Hi @khorne3 -  the documentation states that Semgrep scans reflect the Pro Engine options configured in the UI, but currently, the only visible Pro feature on the settings page is the Cross-File Analysis toggle, there is no option for Pro Engine to be Enabled. This might be misleading, as it implies there are additional configurable options.

Reviewer 2: I think that calls for this to be updated, not removed!

## Pro analysis

> When using `semgrep ci` with Semgrep AppSec Platform, you can configure whether the scan uses cross-file analysis in [Settings](https://semgrep.dev/orgs/-/settings). If you enable cross-file analysis, Semgrep performs interfile and interprocedural analysis for [supported languages](/supported-languages#language-maturity-summary).

Reviewer 2: @khorne3 This feels weird to me because we're switching between cross-file and interfile. I'm not totally clear on when we prefer one vs the other, but what about this:
```suggestion
When using `semgrep ci` with Semgrep AppSec Platform, you can configure whether the scan uses cross-file analysis in [Settings](https://semgrep.dev/orgs/-/settings). If you enable cross-file analysis, Semgrep performs cross-file and cross-function analysis for [supported languages](/supported-languages#language-maturity-summary).
```

Reviewer: I did a quick search of the repo, and we prefer cross-file (though sometimes we'll do something like `cross-file (interfile)`) -- I like your suggestion to use `cross-file` here.

If cross-file analysis is not enabled in Semgrep AppSec Platform, [Pro rules](/docs/semgrep-code/pro-rules) are used, but they are run using cross-function analysis within single files.

> To perform a CLI scan using cross-file-analysis, ensure you've used `install-semgrep-pro` to [install the additional semgrep binary](/docs/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:

Reviewer: ```suggestion
To perform a CLI scan using cross-file-analysis, ensure you've run `semgrep install-semgrep-pro` to [install the additional semgrep binary](/docs/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:
```

Reviewer 2: Oops, I also put an extra hyphen in cross-file analysis here

> To perform a CLI scan using cross-file-analysis, ensure you've run `semgrep install-semgrep-pro` to [install the additional semgrep binary](/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:

Reviewer 2: ```suggestion
To perform a CLI scan using cross-file analysis, ensure you've run `semgrep install-semgrep-pro` to [install the additional semgrep binary](/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:
```

## Pro analysis

When using `semgrep ci` with Semgrep AppSec Platform, you can configure whether the scan uses cross-file analysis in [Settings](https://semgrep.dev/orgs/-/settings). If you enable cross-file analysis, Semgrep performs cross-file and cross-function analysis for [supported languages](/supported-languages#language-maturity-summary).

If cross-file analysis is not enabled in Semgrep AppSec Platform, [Pro rules](/docs/semgrep-code/pro-rules) are used, but they are run using cross-function analysis within single files.

To perform a CLI scan using cross-file analysis, ensure you've run `semgrep install-semgrep-pro` to [install the additional semgrep binary](/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:

```bash
semgrep --config auto --pro
```

To disable cross-file analysis in CI while still using Pro Engine, use:

```bash
semgrep ci --pro-intrafile
```

> If you want to fully revert to OSS-only analysis, disregarding Pro Engine entirely, use:

Reviewer 2: I think we could really omit this part as it doesn't help folks with matching their results, but I don't feel strongly about it.

## Pro analysis

> When using `semgrep ci` with the Semgrep AppSec Platform, the Semgrep scan reflects the Pro Engine capabilities available in Semgrep AppSec Platform at https://semgrep.dev/orgs/-/settings. The only configurable Pro Engine feature in the UI is Cross-File Analysis. Cross-File Analysis if enabled, Semgrep performs interfile and interprocedural analysis for supported Pro languages.

Reviewer: ```suggestion
When using `semgrep ci` with Semgrep AppSec Platform, you can configure whether the scan uses cross-file analysis in [Settings](https://semgrep.dev/orgs/-/settings). If you enable cross-file analysis, Semgrep performs interfile and interprocedural analysis for supported Pro languages.
```

Reviewer 2: Do we still use the term "Pro languages", or would plain "for supported languages" be better here?

Reviewer: Good call -- we can use supported languages and link to https://semgrep.dev/docs/supported-languages#language-maturity-summary (I can add the link)

Reviewer 2: nice!

> If Pro Engine is not enabled for Semgrep scans, [Pro rules](/docs/semgrep-code/pro-rules) are used, but they are run as OSS rules, using only intrafile and intraprocedural analysis.

Reviewer 2: ```suggestion
If cross-file analysis is not enabled in Semgrep AppSec Platform, [Pro rules](/docs/semgrep-code/pro-rules) are used, but they are run using cross-function analysis within single files.
```

> To perform a CLI scan using Pro Engine, ensure you've [installed Pro Engine](/docs/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:

Reviewer 2: ```suggestion
To perform a CLI scan using cross-file-analysis, ensure you've used `install-semgrep-pro` to [install the additional semgrep binary](/docs/semgrep-code/semgrep-pro-engine-intro/#run-cross-file-analysis-in-the-cli), and include `--pro` in your command:
```
